File: eTravelAgencija.Model/model/Hotel.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using eTravelAgencija.Model.model;

namespace eTravelAgencija.Model.model
{
    public partial class Hotel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public int Stars { get; set; }
        public decimal CalculatedPrice { get; set; }
        public virtual ICollection<HotelRooms> HotelRooms { get; set; } = new List<HotelRooms>();
        public virtual ICollection<HotelImages> HotelImages { get; set; } = new List<HotelImages>();
        public virtual ICollection<OfferHotels> OfferHotels { get; set; } = new List<OfferHotels>();
    }
}

File: UI/etravel_mobile/lib/models/hotel.dart
import 'package:json_annotation/json_annotation.dart';
import 'hotel_room.dart';
import 'hotel_image.dart';
import 'offer_hotel.dart';

part 'hotel.g.dart';

@JsonSerializable(explicitToJson: true)
class Hotel {
  final int id;
  final String name;
  final String address;
  final int stars;
  final double calculatedPrice;

  final List<HotelRoom> hotelRooms;
  final List<HotelImage> hotelImages;
  final List<OfferHotel> offerHotels;

  Hotel({
    required this.id,
    required this.name,
    required this.address,
    required this.stars,
    required this.calculatedPrice,
    required this.hotelRooms,
    required this.hotelImages,
    required this.offerHotels,
  });

  factory Hotel.fromJson(Map<String, dynamic> json) => _$HotelFromJson(json);
  Map<String, dynamic> toJson() => _$HotelToJson(this);
}
